core/src/main/java/com/netflix/conductor/core/execution/tasks/SystemTaskWorkerCoordinator.java

```diff
@@ -120,39 +139,95 @@ private void listen() {
                 return;
 			}
 
-			List<String> polledTaskIds = queueDAO.pop(taskName, realPollCount, 200);
-			Monitors.recordTaskPoll(taskName);
-			logger.debug("Polling for {}, got {} tasks", taskName, polledTaskIds.size());
+			List<String> polledTaskIds = queueDAO.pop(queueName, realPollCount, 200);
+			Monitors.recordTaskPoll(queueName);
+			logger.debug("Polling for {}, got {} tasks", queueName, polledTaskIds.size());
 			for(String taskId : polledTaskIds) {
-				logger.debug("Task: {} of type: {} being sent to the workflow executor", taskId, taskName);
+				logger.debug("Task: {} from queue: {} being sent to the workflow executor", taskId, queueName);
 				try {
-					executorService.submit(()-> workflowExecutor.executeSystemTask(systemTask, taskId, unackTimeout));
+					String taskName = QueueUtils.getTaskType(queueName);
+					WorkflowSystemTask systemTask = taskNameWorkFlowTaskMapping.get(taskName);
+					ExecutorService executorService = executionConfig.service;
+					executorService.submit(() -> workflowExecutor.executeSystemTask(systemTask, taskId, unackTimeout));
 				} catch(RejectedExecutionException ree) {
-					logger.warn("Queue full for workers. Size: {}, task:{}", workerQueue.size(), taskName);
+					logger.warn("Queue full for workers. Size: {}, queue:{}", workerQueue.size(), queueName);
 				}
 			}
 		} catch (Exception e) {
 			Monitors.error(className, "pollAndExecute");
-			logger.error("Error executing system task:{}", taskName, e);
+			logger.error("Error executing system task in queue:{}", queueName, e);
+		}
+	}
+
+
+	public boolean isFromCoordinatorDomain(String queueName) {
+		String queueDomain = QueueUtils.getQueueDomain(queueName);
+		return StringUtils.equals(queueDomain, this.domain);
+	}
+
+	private boolean shouldListen(String workflowSystemTaskQueueName) {
+		return isFromCoordinatorDomain(workflowSystemTaskQueueName) && isSystemTask(workflowSystemTaskQueueName);
+	}
+
+
+	public static boolean isSystemTask(String queue) {
+
+		String taskType = QueueUtils.getTaskType(queue);
+
+		if(StringUtils.isNotBlank(taskType)) {
+
+			WorkflowSystemTask task = taskNameWorkFlowTaskMapping.get(taskType);
+			return Objects.nonNull(task) && task.isAsync();
+
 		}
+
+		return false;
+	}
+
+
+
+	public ExecutionConfig getExecutionConfig(String taskQueue) {
+
+		if (!QueueUtils.isIsolatedQueue(taskQueue)) {
+			return this.defaultExecutionConfig;
+		}
+
+		return queueExecutionConfigMap.computeIfAbsent(taskQueue, __ -> this.createExecutionConfig());
+
+	}
+
+	private ExecutionConfig createExecutionConfig() {
+
+		int workerQueueSize = config.getIntProperty("workflow.isolated.system.task.worker.queue.size", 100);
+		LinkedBlockingQueue<Runnable> workerQueue = new LinkedBlockingQueue<>(workerQueueSize);
+		int threadCount = config.getIntProperty("workflow.isolated.system.task.worker.thread.count", 10);
```

Comment:
Since, this is the number of threads in the pool for each isolation group, I think its better to be defensive here and set a default of 1.
The `defaultExecutionConfig` which is used by the non-isolated groups can be configured to 10 threads in its thread pool.